src/Builders/ComponentParameterBuilder.cs

```diff
@@ -1,157 +1,175 @@
+        public ComponentParameterBuilder<TComponent> Add(Expression<Func<TComponent, EventCallback>> parameterSelector, Func<Task> callback)
+        {
+            if (parameterSelector == null)
+            {
+                throw new ArgumentNullException(nameof(parameterSelector));
+            }
+            if (callback == null)
+            {
+                throw new ArgumentNullException(nameof(callback));
+            }
+
+            return Add(parameterSelector, EventCallback.Factory.Create(this, callback));
+        }
+
+        /// <summary>
+        /// Add a generic <see cref="EventCallback"/> parameter with a value for this builder.
+        /// </summary>
+        /// <param name="parameterSelector">The parameter selector</param>
+        /// <param name="callback">The event callback.</param>
+        /// <returns>A <see cref="ComponentParameterBuilder&lt;TComponent&gt;"/> which can be chained.</returns>
+        public ComponentParameterBuilder<TComponent> Add(Expression<Func<TComponent, EventCallback<EventArgs>>> parameterSelector, Func<EventArgs, Task> callback)
+        {
+            if (parameterSelector == null)
+            {
+                throw new ArgumentNullException(nameof(parameterSelector));
+            }
+            if (callback == null)
+            {
+                throw new ArgumentNullException(nameof(callback));
+            }
+
+            return Add(parameterSelector, EventCallback.Factory.Create(this, callback));
+        }
+
+        /// <summary>
+        /// Create a <see cref="ComponentParameter"/> collection.
+        /// </summary>
+        /// <returns>A IReadOnlyCollection of <see cref="ComponentParameter"/></returns>
+        public IReadOnlyCollection<ComponentParameter> Build()
+        {
+            return _componentParameters.ToArray();
+        }
+
+        private static (string name, bool isCascading) GetDetailsFromExpression<T>(Expression<Func<TComponent, T>> parameterSelector)
+        {
+            if (parameterSelector == null)
+            {
+                throw new ArgumentNullException(nameof(parameterSelector));
+            }
+
+            if (parameterSelector.Body is MemberExpression memberExpression && memberExpression.Member is PropertyInfo propertyInfo)
+            {
+                var parameterAttribute = propertyInfo.GetCustomAttribute<ParameterAttribute>();
+                var cascadingParameterAttribute = propertyInfo.GetCustomAttribute<CascadingParameterAttribute>();
+                if (parameterAttribute is null && cascadingParameterAttribute is null)
+                {
+                    throw new ArgumentException($"The parameterSelector '{parameterSelector}' does not have the [Parameter] or [CascadingParameter] attribute set on the property '{propertyInfo.Name}' in the class '{typeof(TComponent)}'.");
+                }
+
+                if (parameterAttribute != null)

```

Comment:
```suggestion
                if (parameterAttribute is {})
```
I use `xxx is {}` in the library consistently to check for `not null`.